src/lib/storage/fixed_string_dictionary_segment.cpp

```diff
@@ -16,11 +17,75 @@ FixedStringDictionarySegment<T>::FixedStringDictionarySegment(
     const std::shared_ptr<const FixedStringVector>& dictionary,
     const std::shared_ptr<const BaseCompressedVector>& attribute_vector)
     : BaseDictionarySegment(data_type_from_type<pmr_string>()),
-      _dictionary{dictionary},
-      _dictionary_span{std::make_shared<const FixedStringSpan>(*_dictionary)},
+      _dictionary_base_vector{dictionary},
+      _dictionary{std::make_shared<const FixedStringSpan>(*_dictionary_base_vector)},
       _attribute_vector{attribute_vector},
       _decompressor{_attribute_vector->create_base_decompressor()} {}
 
+template <typename T>
+FixedStringDictionarySegment<T>::FixedStringDictionarySegment(const std::byte* start_address)
+    : BaseDictionarySegment(data_type_from_type<T>()) {
+  const auto header_data = reinterpret_cast<const uint32_t*>(start_address);
+  const auto encoding_type = PersistedSegmentEncodingType{header_data[ENCODING_TYPE_OFFSET_INDEX]};
+  const auto string_length = header_data[STRING_LENGTH_OFFSET_INDEX];
+  const auto dictionary_size = header_data[DICTIONARY_SIZE_OFFSET_INDEX];
+  const auto attribute_vector_size = header_data[ATTRIBUTE_VECTOR_OFFSET_INDEX];
+
+  const auto* dictionary_address = reinterpret_cast<const char*>(start_address + HEADER_OFFSET_BYTES);
+  const auto dictionary_span_pointer =
+      std::make_shared<const FixedStringSpan>(dictionary_address, string_length, dictionary_size);
+  //TODO: Move to size_bytes function on FixedStringSpan
+  const auto dictionary_size_bytes = dictionary_size * string_length;
+
+  switch (encoding_type) {
+    case PersistedSegmentEncodingType::Unencoded: {
+      Fail("Unencoded Segments are not yet implemented for mmap-based storage.");
+      break;
+    }
+    case PersistedSegmentEncodingType::DictionaryEncoding8Bit: {
+      auto* const attribute_vector_address =
+          reinterpret_cast<const uint8_t*>(start_address + HEADER_OFFSET_BYTES + dictionary_size_bytes);
+      auto attribute_data_span = std::span<const uint8_t>(attribute_vector_address, attribute_vector_size);
+      auto attribute_vector = std::make_shared<FixedWidthIntegerVector<uint8_t>>(attribute_data_span);
+
+      _dictionary = dictionary_span_pointer;
+      _attribute_vector = attribute_vector;
+      _decompressor = _attribute_vector->create_base_decompressor();
+      break;
+    }
+    case PersistedSegmentEncodingType::DictionaryEncoding16Bit: {
+      auto* const attribute_vector_address =
```

Comment:
Not to be addressed: There is a lot of duplication here and in the Dictionary class. That could  have been put into little helpers. There are also resolve patterns that would have helped to get rid of a lot of code here.